src/widgetastic_patternfly4/bulletchart.py

```diff
@@ -0,0 +1,166 @@
+    )
+    LEGEND_TEXT = ".//*[name()='text' and contains(@id, 'legend-labels')]"
+    LEGEND_ITEM_REGEX = re.compile(r"(\d+)\s(\w.*)|(\w.*)\s(\d+)")
+
+    ITEMS = ".//*[name()='g']/*[name()='path' and not(contains(@style, 'type:square'))]"
+    TOOLTIP_REGEX = re.compile(r"(.*?): ([\d]+)")
+    APPLY_OFFSET = True
+
+    tooltip = Text(
+        ".//*[name()='svg' and contains(@aria-labelledby, 'victory-container')]/"
+        "following-sibling::div[contains(@style, 'z-index')]/*[name()='svg']"
+    )
+
+    def __init__(self, parent=None, id=None, locator=None, logger=None, *args, **kwargs):
+        View.__init__(self, parent=parent, logger=logger)
+        if id:
+            self.locator = ".//div[@id={}]".format(quote(id))
+        elif locator:
+            self.locator = locator
+        else:
+            self.locator = self.DEFAULT_LOCATOR
+
+        self.args = args
+        self.kwargs = kwargs
+
+    def _offsets(self, el):
+        """Calculate offset. Need to set offset with try and error method."""
+        offset_denominator = self.kwargs.pop("offset_denominator", 2.5)
+        size = self.browser.size_of(el)
+        width = size.width  # el.size["width"]
+        height = size.height  # el.size["height"]
+        dx = int(width / offset_denominator) if (width > 10 and height > 10) else 0
+        dy = int(height / offset_denominator) if (width > 10 and height > 10) else 0
+        return dx, dy
+
+    @property
+    def legends(self):
+        """Return all Legend objects."""
+
+        br = self.browser
+        _data = []
+        for (icon, label_el) in zip(br.elements(self.LEGEND_ICON), br.elements(self.LEGEND_TEXT)):
+            label_text = br.text(label_el)
+            match = self.LEGEND_ITEM_REGEX.match(label_text)
+            if match:
+                left_value, left_label, right_label, right_value = match.groups()
+            label = right_label or left_label if match else label_text
+            value = int(right_value or left_value) if match else None
+            _data.append(
+                Legend(
+                    label=label,
+                    value=value,
+                    color=icon.value_of_css_property("fill"),
+                    element=label_el,
+                )
+            )
+        return _data
+
+    @property
+    def legend_names(self):
+        """Return all legend names."""
+        return [leg.label for leg in self.legends]
+
+    def get_legend(self, label):
+        """Get specific Legend object.
+        Arg:
+            label: Name of legend label.
+        """
+        try:
+            return next(leg for leg in self.legends if leg.label == label)
+        except StopIteration:
+            return None
+
+    @property
+    def data(self):
+        """Read graph and returns all Bar objects."""
+        _data = []
+        # focus away from graph
+        self.parent_browser.raw_click(".//body")
```

Comment:
Are you sure that will work? `.//body` means that you look for `body` element in the current `ROOT`. You might want to look an absolute path `//body`.

Reply:
my bad. Actually planning to use `move_to_element`